------------
------------

Nettoyage de la colonne Place

In [ ]:
import pandas as pd

df_inscriptions = pd.read_csv('/content/cyprus-inscriptions.csv', sep=',', encoding='utf-8-sig', engine='python')
df_inscriptions

Inscription # Type  \
0                                       ICS #456: p.389  CS?   
1                                       ICS #419: p.373   CS   
2                                    ICS #392: pp.363-4   CS   
3                                       ICS #388: p.362  EC?   
4                                       ICS #391: p.363   CS   
...                                                 ...  ...   
3800  Enkomi (Schaeffer) 1959 #4\r=Les cylindres-sce...  CM?   
3801        Louvre AM 1639\r=BCH 81 (1957) #9: pp.14-15   CM   
3802       Kourion B1625\r=Bamboula at Kourion, p.144\r  CM?   
3803                K-AD #448\r=K-AD II, p.40, fig.63:6  CM?   
3804                K-AD #449\r=K-AD II, p.40, fig.63:7  CM?   

                                       Place                  Date  \
0                                  Cyrenaica                   NaN   
1      Egypt: Abydos \r(temple of Ramses II)  Classical\rearly 4th   
2         Egypt: Abydos \r(temple of Seti I)  Classical\rearly 4th   
3         Egypt: Abydos \r(temple of Seti I)  Classical\rearly 4th   
4         Egypt: Abydos \r(temple of Seti I)  Classical\rearly 4th   
...                                      ...                   ...   
3800                       Famagusta: Enkomi                 LC 3A   
3801                        Famagusta: Sinda           early LC 3A   
3802          Limassol: Kourion \r"Bamboula"                 LC 3A   
3803   Larnaca: Kalavasos "Ayios Dhimitrios"                 LC 2C   
3804  Larnaca: Kalavasos\r"Ayios Dhimitrios"                 LC 2C   

                                                 Nature  \
0     No interpretation: ancient (archaizing) imitat...   
1                          PN: Timos \rson of Thursiya.   
2                        PN: Menokretes \rfrom Salamis.   
3                                  PN: an Eteo-Cypriot?   
4                                                   PN?   
...                                                 ...   
3800                                            1 sign?   
3801                                            1 sign.   
3802                        1 sign? (secondary carving)   
3803                                            1 sign.   
3804                                           1 sign.    

                                           Material  
0                                Serpentine scarab.  
1                          Stone wall.  \rGraffito.  
2                            Stone wall.\rGraffito.  
3                          Stone wall.  \rGraffito.  
4                           Stone wall. \rGraffito.  
...                                             ...  
3800  Cylinder seal:\rbrown steatite or serpentine.  
3801                       Cylinder seal: hematite.  
3802                      Cylinder seal:\rsteatite.  
3803                                  Metal weight.  
3804                                  Metal weight.  

[3805 rows x 6 columns]

In [ ]:
# Nettoyer le DataFrame pour les noms de lieux
split_1 = df_inscriptions['Place'].str.split(':', n=1, expand=True)
df_inscriptions['Place_Region'] = split_1[0].str.strip()
df_inscriptions['Place_Site_Raw'] = split_1[1].str.strip()
split_2 = df_inscriptions['Place_Site_Raw'].str.split(r'[\(\"]', n=1, expand=True)
df_inscriptions['Place_Site'] = split_2[0].str.strip()
df_inscriptions['Place_precision'] = split_2[1].str.replace(r'[\)\"]', '', regex=True).str.strip()
df_inscriptions.drop(columns=['Place_Site_Raw'], inplace=True)
df_inscriptions.to_csv('cyprus-inscriptions-clean.csv', index=False)
print(df_inscriptions.value_counts('Place_Region'))
print(df_inscriptions.value_counts('Place_Site'))
print(df_inscriptions.value_counts('Place_precision'))

Place_Region
Paphos               967
Limassol             954
Nicosia              604
Famagusta            527
Larnaca              404
Cyprus               139
Egypt                115
Kyrenia               45
Paphos?               15
Syria                  7
Limassol?              5
Cyprus?                4
Greece                 4
Larnaca?               3
Limassol ?             3
Lebanon                2
Cyrenaica              1
Famagusta?             1
Israel                 1
Italy                  1
Nicosia?               1
Nicosia? Larnaca?      1
Syria?                 1
Name: count, dtype: int64
Place_Site
Kafizin                  340
Amathus\rAyios Tychon    256
Kouklia,\rSiege Mound    228
Larnaca \rKition         207
Rantidi                  173
                        ... 
Yialia                     1
Yeroskipou                 1
Alambra                    1
Achelia                    1
Aknha                      1
Name: count, Length: 374, dtype: int64
Place_precision
t

On voit qu'il n'y a pas énormément de noms de région, ce qui est bien pour nous. Il conviendra de nettoyer quelques valeurs. La pratique rigoureuse sera de drop les valeurs avec un point d'interrogation du DataFrame.
Quant aux sites, j'affiche les valeurs uniques pour voir si on ne peut pas les normaliser.

In [ ]:
df_inscriptions = df_inscriptions[~df_inscriptions['Place'].str.contains(r'\?', na=False)]
df_inscriptions.value_counts('Place_Region')

Place_Region
Paphos       940
Limassol     936
Nicosia      588
Famagusta    510
Larnaca      398
Cyprus       139
Egypt        115
Kyrenia       43
Syria          6
Greece         4
Lebanon        2
Cyrenaica      1
Israel         1
Italy          1
Name: count, dtype: int64

In [ ]:
df_inscriptions['Place_Site'].unique()

array([None, 'Abydos', 'Ehnasya', 'Gebel Abou Fodah', 'Gizeh', 'Karnak',
       'Naukratis', 'Nubia, Bouhen', 'Shatt El-Saba El Rigal', 'Thebes',
       'Athens', 'Delphi', 'Rhodes', 'Tel Dor', 'Policoro', 'Sarepta',
       'Sidon', 'Ras Shamra', 'Akanthou', 'Akhna', 'Aknha', 'Akrotiri',
       'Alep, \rnear Salamis', 'Angastina', 'Arsos', 'Ayia Triada',
       'Ayios Varnavas', 'Drosistis, \rnear Sinda', 'Enkomi', 'Famagusta',
       'Gypsos, near Salamis.', 'Knodhara, near Salamis',
       'Koma tou Yalou', 'Kouklia, near Salamis', 'Limnia', 'Magrastica',
       'Marathovounos', 'near Rhizokarpaso', 'Phrenaros, near Famagusta',
       'Rhizokarpasso', 'Rizokarpaso', 'Salamis', 'Salamis,\ragora',
       'Salamis, Basilica', 'Salamis, gymnasium',
       'Salamis, gymnasium baths', 'Salamis, sanctuary of Zeus',
       'Salamis,\rtheater', 'Sklinitzia, near Salamis',
       'Syncrasis, near Salamis', 'Yalousa', 'Ayia Eirene',
       'Ayios Ambrosios', 'between Ayios Epiktitos & Bellapais

In [ ]:
df_inscriptions['Place_Site'] = df_inscriptions['Place_Site'].str.replace(r'[\r\n]+', ' ', regex=True).str.strip()
df_inscriptions['Place_Site'] = df_inscriptions['Place_Site'].str.replace(r'\s+', ' ', regex=True)
df_inscriptions['Place_Site'].unique()
# J'enlève tout ce qui commence par "near" pour n'avoir que des données précises
df_inscriptions = df_inscriptions[~df_inscriptions['Place_Site'].str.startswith('near').fillna(False)]
df_inscriptions['Place_Site'].unique()

/tmp/ipython-input-812433570.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_inscriptions['Place_Site'] = df_inscriptions['Place_Site'].str.replace(r'[\r\n]+', ' ', regex=True).str.strip()
/tmp/ipython-input-812433570.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_inscriptions['Place_Site'] = df_inscriptions['Place_Site'].str.replace(r'\s+', ' ', regex=True)
/tmp/ipython-input-812433570.py:5: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will chan

array([None, 'Abydos', 'Ehnasya', 'Gebel Abou Fodah', 'Gizeh', 'Karnak',
       'Naukratis', 'Nubia, Bouhen', 'Shatt El-Saba El Rigal', 'Thebes',
       'Athens', 'Delphi', 'Rhodes', 'Tel Dor', 'Policoro', 'Sarepta',
       'Sidon', 'Ras Shamra', 'Akanthou', 'Akhna', 'Aknha', 'Akrotiri',
       'Alep, near Salamis', 'Angastina', 'Arsos', 'Ayia Triada',
       'Ayios Varnavas', 'Drosistis, near Sinda', 'Enkomi', 'Famagusta',
       'Gypsos, near Salamis.', 'Knodhara, near Salamis',
       'Koma tou Yalou', 'Kouklia, near Salamis', 'Limnia', 'Magrastica',
       'Marathovounos', 'Phrenaros, near Famagusta', 'Rhizokarpasso',
       'Rizokarpaso', 'Salamis', 'Salamis, agora', 'Salamis, Basilica',
       'Salamis, gymnasium', 'Salamis, gymnasium baths',
       'Salamis, sanctuary of Zeus', 'Salamis, theater',
       'Sklinitzia, near Salamis', 'Syncrasis, near Salamis', 'Yalousa',
       'Ayia Eirene', 'Ayios Ambrosios',
       'between Ayios Epiktitos & Bellapais', 'Dhiorios',
       'Harm

In [ ]:
mapping_corrections = {
    'Aknha': 'Akhna',
    'Dhekhelia': 'Dhekelia',
    'LimassoL': 'Limassol',
    'Rhizokarpasso': 'Rhizokarpaso',
    'Tremethousha': 'Tremetousha',
    'Apheshia': 'Apeshia',
    'Pano Polimidhia': 'Pano Polemidia'
}
df_inscriptions['Place_Site'] = df_inscriptions['Place_Site'].replace(mapping_corrections)
df_inscriptions['Place_Site'].unique()

array([None, 'Abydos', 'Ehnasya', 'Gebel Abou Fodah', 'Gizeh', 'Karnak',
       'Naukratis', 'Nubia, Bouhen', 'Shatt El-Saba El Rigal', 'Thebes',
       'Athens', 'Delphi', 'Rhodes', 'Tel Dor', 'Policoro', 'Sarepta',
       'Sidon', 'Ras Shamra', 'Akanthou', 'Akhna', 'Akrotiri',
       'Alep, near Salamis', 'Angastina', 'Arsos', 'Ayia Triada',
       'Ayios Varnavas', 'Drosistis, near Sinda', 'Enkomi', 'Famagusta',
       'Gypsos, near Salamis.', 'Knodhara, near Salamis',
       'Koma tou Yalou', 'Kouklia, near Salamis', 'Limnia', 'Magrastica',
       'Marathovounos', 'Phrenaros, near Famagusta', 'Rhizokarpaso',
       'Rizokarpaso', 'Salamis', 'Salamis, agora', 'Salamis, Basilica',
       'Salamis, gymnasium', 'Salamis, gymnasium baths',
       'Salamis, sanctuary of Zeus', 'Salamis, theater',
       'Sklinitzia, near Salamis', 'Syncrasis, near Salamis', 'Yalousa',
       'Ayia Eirene', 'Ayios Ambrosios',
       'between Ayios Epiktitos & Bellapais', 'Dhiorios',
       'Harmanlar, nea

In [ ]:
# Supprimer "T." suivi de chiffres, avec ou sans espaces, et éventuellement une lettre (ex: T.A128)
df_inscriptions['Place_Site'] = (df_inscriptions['Place_Site']
                                 .str.replace(r',?\s*T\s*\.?\s*[A-Z]?\d+\w*', '', regex=True)
                                 .str.strip())
print(df_inscriptions['Place_Site'].unique())
df_inscriptions.value_counts('Place_Site')

[None 'Abydos' 'Ehnasya' 'Gebel Abou Fodah' 'Gizeh' 'Karnak' 'Naukratis'
 'Nubia, Bouhen' 'Shatt El-Saba El Rigal' 'Thebes' 'Athens' 'Delphi'
 'Rhodes' 'Tel Dor' 'Policoro' 'Sarepta' 'Sidon' 'Ras Shamra' 'Akanthou'
 'Akhna' 'Akrotiri' 'Alep, near Salamis' 'Angastina' 'Arsos' 'Ayia Triada'
 'Ayios Varnavas' 'Drosistis, near Sinda' 'Enkomi' 'Famagusta'
 'Gypsos, near Salamis.' 'Knodhara, near Salamis' 'Koma tou Yalou'
 'Kouklia, near Salamis' 'Limnia' 'Magrastica' 'Marathovounos'
 'Phrenaros, near Famagusta' 'Rhizokarpaso' 'Rizokarpaso' 'Salamis'
 'Salamis, agora' 'Salamis, Basilica' 'Salamis, gymnasium'
 'Salamis, gymnasium baths' 'Salamis, sanctuary of Zeus'
 'Salamis, theater' 'Sklinitzia, near Salamis' 'Syncrasis, near Salamis'
 'Yalousa' 'Ayia Eirene' 'Ayios Ambrosios'
 'between Ayios Epiktitos & Bellapais' 'Dhiorios'
 'Harmanlar, near Kyrenia' 'Kazaphani' 'Kyrenia' 'Lapithos'
 'Melandrina, near Ayios Ambrosios' 'Asgata' 'Athienou Golgoi' 'Dhekelia'
 'Gai, near Dromolaxia' 'Hala Sul

Place_Site
Kafizin                                340
Amathus Ayios Tychon                   256
Kouklia, Siege Mound                   228
Larnaca Kition                         208
Rantidi                                173
                                      ... 
Achelia                                  1
Yeroskipou                               1
Yialia                                   1
between Ayios Epiktitos & Bellapais      1
south of Evdhimos, near Kourion          1
Name: count, Length: 270, dtype: int64

Au final, je décide de construire un dictionnaire un peu agressif, qui perd certaines précisions, car je travaille ici à l'échelle de l'île.

In [ ]:
normalization_dict = {
    # --- AMATHUS ---
    'Amathus acropolis': 'Amathus',
    'Amathus acropolis sanct. Aphrodite': 'Amathus',
    'Amathus acropolis sanctuary Aphrodite': 'Amathus',
    'Amathus acropolis, sanctuary of Aphrodite': 'Amathus',
    'Amathus Armenochori': 'Amathus',
    'Amathus Ayios Tychon': 'Amathus',
    'Amathus E necropolis': 'Amathus',
    'Amathus Khani tis Monis': 'Amathus',
    'Amathus N necropolis': 'Amathus',
    'Amathus Parekklesia': 'Amathus',
    'Amathus SW necropolis': 'Amathus',
    'Amathus W necropolis': 'Amathus',
    'Amathus, acropolis': 'Amathus',
    'near Ayios Tychon': 'Amathus',

    # --- KOURION ---
    'Kourion acropolis': 'Kourion',
    'Kourion, acropolis': 'Kourion',
    'Kourion, acropolis: Basilica': 'Kourion',
    'Kourion, acropolis: near Basilica': 'Kourion',
    'Kourion Episkopi': 'Kourion',
    'Kourion House of the Achilleus Mosaic': 'Kourion',
    'Kourion near acropolis': 'Kourion',
    'Kourion near Basilica': 'Kourion',
    'Kourion sanct. Demeter & Kore': 'Kourion',
    'Kourion sanctuary of Apollo': 'Kourion',
    'Kourion near temple of Apollo': 'Kourion',
    'Kourion sanctuary of Demeter & Kore': 'Kourion',
    'Kourion temple of Hylate': 'Kourion',
    'Kourion theater': 'Kourion',
    'Kourion Villa of the Gladiators': 'Kourion',
    'Kourion Yerakarka': 'Kourion',
    'Kourion Altar 1': 'Kourion',
    'Ayios Hermogenis, near Kourion': 'Kourion',
    'south of Evdhimos, near Kourion': 'Kourion',
    'near Kourion': 'Kourion',
    'Episkopi': 'Kourion',

    # --- SALAMIS ---
    'Salamis, agora': 'Salamis',
    'Salamis, Basilica': 'Salamis',
    'Salamis, gymnasium': 'Salamis',
    'Salamis, gymnasium baths': 'Salamis',
    'Salamis, sanctuary of Zeus': 'Salamis',
    'Salamis, theater': 'Salamis',
    'Salamis necropolis': 'Salamis',
    'Alep, near Salamis': 'Salamis',
    'Gypsos, near Salamis.': 'Salamis',
    'Knodhara, near Salamis': 'Salamis',
    'Kouklia, near Salamis': 'Salamis',
    'Sklinitzia, near Salamis': 'Salamis',
    'Syncrasis, near Salamis': 'Salamis',
    'Constanzia, near Salamis': 'Salamis',

    # --- KOUKLIA (PALAEPAPHOS) ---
    'Kouklia, temple area': 'Kouklia',
    'Kouklia KD': 'Kouklia',
    'Kouklia TE': 'Kouklia',
    'Kouklia KB': 'Kouklia',
    'Kouklia KC, temple area': 'Kouklia',
    'Kouklia, Siege Mound': 'Kouklia',
    'Ayios Epiphanios, near Kouklia': 'Kouklia',
    'Kato Alonia, near Kouklia': 'Kouklia',
    'Argaro, near Kouklia': 'Kouklia',
    'Eleomylia, near Kouklia': 'Kouklia',
    'Hassan Agha, near Kouklia': 'Kouklia',
    'Khapotami river, near Kouklia': 'Kouklia',
    'Mantissa, near Kouklia': 'Kouklia',
    'Palaeomylos, near Kouklia': 'Kouklia',
    'Xylinos, near Kouklia': 'Kouklia',
    'Palaepaphos': 'Kouklia',
    'near Kouklia': 'Kouklia',

    # --- POLIS (MARION / ARSINOE) ---
    'Polis Arsinoe': 'Polis',
    'Polis Marion': 'Polis',
    'Polis Marion/ Arsinoe': 'Polis',
    'Polis Marion/ Arsinoe, Necropolis I': 'Polis',
    'Polis Marion/ Arsinoe,': 'Polis',
    'Polis Marion, Site A': 'Polis',
    'Polis Marion/ Arsinoe, Site A': 'Polis',
    'Polis Marion/ Arsinoe, Site B': 'Polis',
    'Polis Marion, Site C': 'Polis',
    'Polis Marion, Site I': 'Polis',
    'Polis Marion/ Arsinoe, Site I': 'Polis',
    'Polis Marion/ Arsinoe, Site II': 'Polis',
    'Polis Marion/ Arsinoe, Site III': 'Polis',
    'Polis Marion/ Arsinoe, Site K': 'Polis',
    'Polis Marion, Site K': 'Polis',
    'Polis Marion/ Arsinoe, Site M': 'Polis',
    'Steni, near Polis': 'Polis',

    # --- LARNACA (KITION) ---
    'Larnaca Kition': 'Larnaca',
    'Larnaca, sanctuary of Artemis Paralia': 'Larnaca',
    'Kition': 'Larnaca',

    # --- PAPHOS (NEA PAPHOS) ---
    'Kato Paphos': 'Paphos',
    'Nea Paphos': 'Paphos',
    'Nea Paphos, House of Dionysos': 'Paphos',

    # --- DHALI (IDALION) ---
    'Dhali Idalion': 'Dhali',
    'Dhali Idalion sanctuary of Athena': 'Dhali',
    'Dhali Idalion temple of Apollo': 'Dhali',
    'near Dhali/Idalion': 'Dhali',

    # --- SOLOI ---
    'Karavostasi, near Soloi': 'Soloi',
    'Mersinaki, near Soloi': 'Soloi',
    'Soloi Potamos tou Kambou': 'Soloi',
    'Soloi Potamos tou Kambou,': 'Soloi',

    # --- AUTRES ---
    'Rizokarpaso': 'Rhizokarpaso',
    'Ayios Athanansios': 'Ayios Athanasios',
    'Polemidhia': 'Pano Polemidia',
    'LimassoL': 'Limassol',
    'Athienou Golgoi': 'Athienou',
    'Politiko Tamassos': 'Tamassos',
    'Koupati, near Ledra': 'Ledra',
    'Gai, near Dromolaxia': 'Dromolaxia',
    'Drosistis, near Sinda': 'Sinda',
    'Vouni, necropolis': 'Vouni',
    'Vouni, temple of Athena': 'Vouni',
    'Vouni, temple of Artemis Paralia': 'Vouni',
    'Phasoulla, sanctuary of Zeus': 'Phasoulla',
    'Phasoulla, temple of Artemis Paralia': 'Phasoulla',
    'Akamas area': 'Akamas',
    'Akamas, temple area': 'Akamas',
    'Khapotami, 600 m. to the west': 'Khapotami',
    'Phrangissa, sanctuary of Apollo': 'Phrangissa',
    'between Ayios Epiktitos & Bellapais': 'Ayios Epiktitos'
}

df_inscriptions['Site'] = df_inscriptions['Place_Site'].replace(normalization_dict)
print(df_inscriptions['Site'].unique())
df_inscriptions.value_counts('Site')

[None 'Abydos' 'Ehnasya' 'Gebel Abou Fodah' 'Gizeh' 'Karnak' 'Naukratis'
 'Nubia, Bouhen' 'Shatt El-Saba El Rigal' 'Thebes' 'Athens' 'Delphi'
 'Rhodes' 'Tel Dor' 'Policoro' 'Sarepta' 'Sidon' 'Ras Shamra' 'Akanthou'
 'Akhna' 'Akrotiri' 'Salamis' 'Angastina' 'Arsos' 'Ayia Triada'
 'Ayios Varnavas' 'Sinda' 'Enkomi' 'Famagusta' 'Koma tou Yalou' 'Limnia'
 'Magrastica' 'Marathovounos' 'Phrenaros, near Famagusta' 'Rhizokarpaso'
 'Yalousa' 'Ayia Eirene' 'Ayios Ambrosios' 'Ayios Epiktitos' 'Dhiorios'
 'Harmanlar, near Kyrenia' 'Kazaphani' 'Kyrenia' 'Lapithos'
 'Melandrina, near Ayios Ambrosios' 'Asgata' 'Athienou' 'Dhekelia'
 'Dromolaxia' 'Hala Sultan Tekke' 'Kalavasos' 'Kellia' 'Kiti' 'Kophinou'
 'Larnaca' 'Laxia tou Solomou' 'Leukolla' 'Louroujina' 'Mari' 'Maroni'
 'Melousha, near Athienou' 'Pyla' 'Scarino' 'Tremetousha' 'Troulloi'
 'Amathus' 'Ano Kyvides' 'Anogyra' 'Apeshia' 'Avdemou' 'Ayios Athanasios'
 'Ayios Georgios' 'Kourion' 'Gerasa' 'Kato Pissouri' 'Khandria'
 'Khirokitia' 'Limassol' 

Site
Amathus          494
Kouklia          468
Kafizin          340
Salamis          306
Larnaca          237
                ... 
Thebes             1
Tochni             1
Tsiflikoudkia      1
Yeroskipou         1
Yialia             1
Name: count, Length: 171, dtype: int64

Après ça, on passe de 270 à 163 valeurs uniques dans Site

In [ ]:
df_inscriptions = df_inscriptions.drop(columns=['Place_Site'])
df_inscriptions

Inscription # Type  \
0                                       ICS #456: p.389  CS?   
1                                       ICS #419: p.373   CS   
2                                    ICS #392: pp.363-4   CS   
3                                       ICS #388: p.362  EC?   
4                                       ICS #391: p.363   CS   
...                                                 ...  ...   
3800  Enkomi (Schaeffer) 1959 #4\r=Les cylindres-sce...  CM?   
3801        Louvre AM 1639\r=BCH 81 (1957) #9: pp.14-15   CM   
3802       Kourion B1625\r=Bamboula at Kourion, p.144\r  CM?   
3803                K-AD #448\r=K-AD II, p.40, fig.63:6  CM?   
3804                K-AD #449\r=K-AD II, p.40, fig.63:7  CM?   

                                       Place                  Date  \
0                                  Cyrenaica                   NaN   
1      Egypt: Abydos \r(temple of Ramses II)  Classical\rearly 4th   
2         Egypt: Abydos \r(temple of Seti I)  Classical\rearly 4th   
3         Egypt: Abydos \r(temple of Seti I)  Classical\rearly 4th   
4         Egypt: Abydos \r(temple of Seti I)  Classical\rearly 4th   
...                                      ...                   ...   
3800                       Famagusta: Enkomi                 LC 3A   
3801                        Famagusta: Sinda           early LC 3A   
3802          Limassol: Kourion \r"Bamboula"                 LC 3A   
3803   Larnaca: Kalavasos "Ayios Dhimitrios"                 LC 2C   
3804  Larnaca: Kalavasos\r"Ayios Dhimitrios"                 LC 2C   

                                                 Nature  \
0     No interpretation: ancient (archaizing) imitat...   
1                          PN: Timos \rson of Thursiya.   
2                        PN: Menokretes \rfrom Salamis.   
3                                  PN: an Eteo-Cypriot?   
4                                                   PN?   
...                                                 ...   
3800                                            1 sign?   
3801                                            1 sign.   
3802                        1 sign? (secondary carving)   
3803                                            1 sign.   
3804                                           1 sign.    

                                           Material Place_Region  \
0                                Serpentine scarab.    Cyrenaica   
1                          Stone wall.  \rGraffito.        Egypt   
2                            Stone wall.\rGraffito.        Egypt   
3                          Stone wall.  \rGraffito.        Egypt   
4                           Stone wall. \rGraffito.        Egypt   
...                                             ...          ...   
3800  Cylinder seal:\rbrown steatite or serpentine.    Famagusta   
3801                       Cylinder seal: hematite.    Famagusta   
3802                      Cylinder seal:\rsteatite.     Limassol   
3803                                  Metal weight.      Larnaca   
3804                                  Metal weight.      Larnaca   

          Place_precision       Site  
0                    None       None  
1     temple of Ramses II     Abydos  
2        temple of Seti I     Abydos  
3        temple of Seti I     Abydos  
4        temple of Seti I     Abydos  
...                   ...        ...  
3800                 None     Enkomi  
3801                 None      Sinda  
3802             Bamboula    Kourion  
3803     Ayios Dhimitrios  Kalavasos  
3804     Ayios Dhimitrios  Kalavasos  

[3677 rows x 9 columns]

In [ ]:
df_pas_vide = df_inscriptions[df_inscriptions['Site'].notna()]
df_pas_vide

Inscription # Type  \
1                                       ICS #419: p.373   CS   
2                                    ICS #392: pp.363-4   CS   
3                                       ICS #388: p.362  EC?   
4                                       ICS #391: p.363   CS   
5                                       ICS #381: p.360   CS   
...                                                 ...  ...   
3800  Enkomi (Schaeffer) 1959 #4\r=Les cylindres-sce...  CM?   
3801        Louvre AM 1639\r=BCH 81 (1957) #9: pp.14-15   CM   
3802       Kourion B1625\r=Bamboula at Kourion, p.144\r  CM?   
3803                K-AD #448\r=K-AD II, p.40, fig.63:6  CM?   
3804                K-AD #449\r=K-AD II, p.40, fig.63:7  CM?   

                                       Place                  Date  \
1      Egypt: Abydos \r(temple of Ramses II)  Classical\rearly 4th   
2         Egypt: Abydos \r(temple of Seti I)  Classical\rearly 4th   
3         Egypt: Abydos \r(temple of Seti I)  Classical\rearly 4th   
4         Egypt: Abydos \r(temple of Seti I)  Classical\rearly 4th   
5         Egypt: Abydos \r(temple of Seti I)  Classical\rearly 4th   
...                                      ...                   ...   
3800                       Famagusta: Enkomi                 LC 3A   
3801                        Famagusta: Sinda           early LC 3A   
3802          Limassol: Kourion \r"Bamboula"                 LC 3A   
3803   Larnaca: Kalavasos "Ayios Dhimitrios"                 LC 2C   
3804  Larnaca: Kalavasos\r"Ayios Dhimitrios"                 LC 2C   

                              Nature  \
1       PN: Timos \rson of Thursiya.   
2     PN: Menokretes \rfrom Salamis.   
3               PN: an Eteo-Cypriot?   
4                                PN?   
5                      PN: Zoophaos.   
...                              ...   
3800                         1 sign?   
3801                         1 sign.   
3802     1 sign? (secondary carving)   
3803                         1 sign.   
3804                        1 sign.    

                                           Material Place_Region  \
1                          Stone wall.  \rGraffito.        Egypt   
2                            Stone wall.\rGraffito.        Egypt   
3                          Stone wall.  \rGraffito.        Egypt   
4                           Stone wall. \rGraffito.        Egypt   
5                            Stone wall.\rGraffito.        Egypt   
...                                             ...          ...   
3800  Cylinder seal:\rbrown steatite or serpentine.    Famagusta   
3801                       Cylinder seal: hematite.    Famagusta   
3802                      Cylinder seal:\rsteatite.     Limassol   
3803                                  Metal weight.      Larnaca   
3804                                  Metal weight.      Larnaca   

          Place_precision       Site  
1     temple of Ramses II     Abydos  
2        temple of Seti I     Abydos  
3        temple of Seti I     Abydos  
4        temple of Seti I     Abydos  
5        temple of Seti I     Abydos  
...                   ...        ...  
3800                 None     Enkomi  
3801                 None      Sinda  
3802             Bamboula    Kourion  
3803     Ayios Dhimitrios  Kalavasos  
3804     Ayios Dhimitrios  Kalavasos  

[3480 rows x 9 columns]

---------
---------

Nettoyage de la colonne Date

In [ ]:
df_inscriptions.value_counts('Date')

Date
Hellenistic\r3rd BC\r(225/18)                    264
Roman\r2nd - 3rd AD                              257
Archaic\rlate 6th                                229
Classical\rearly 4th                             107
Archaic\r6th                                     106
                                                ... 
Hellenistic\rmid 2nd BC\r(160)                     1
Hellenistic\rmid 2nd BC\r(163/58?)                 1
Hellenistic\rmid 2nd BC \r(145)                    1
Hellenistic\rmid 2nd BC \r(145/44)                 1
Classical\rlate 5th - early 4th BC (411- 374)      1
Name: count, Length: 673, dtype: int64

On voit qu'il y a beaucoup de valeurs uniques alors que ça pourrait être résumé très facilement, en mettant tout ce qui est après un retour à la ligne dans une nouvelle colonne par exemple.

In [ ]:
split_data = df_inscriptions['Date'].str.split(r'\r', n=1, expand=True)
if 1 in split_data.columns:
    df_inscriptions['Precision_Date'] = split_data[1].str.replace(r'\r', ' ', regex=True).str.strip()
df_inscriptions['Date'] = split_data[0].str.strip()
df_inscriptions['Precision_Date'] = df_inscriptions['Precision_Date'].fillna('')
df_inscriptions.value_counts('Date')

Date
Roman                       930
Hellenistic                 824
Archaic                     411
Classical                   245
LC                           87
                           ... 
Roman- Byzantine              1
end LC                        1
late Archaic - Classical      1
late Hellenistic/Roman        1
seal = Mitannian style;       1
Name: count, Length: 81, dtype: int64

In [ ]:
print(df_inscriptions['Date'].unique())


[nan 'Classical' 'Archaic' 'LC 3B' 'LC 2C' 'LC 2C - 3A'
 'Classical - Hellenistic' 'Hellenistic' 'late Hellenistic/' 'Roman'
 'Roman?' 'Roman (Hadrianic)' 'Roman (Neronian)' 'late Hellenistic'
 'Roman (late)' 'Roman (Antonine)' 'Roman (Severan)'
 'late Hellenistic/ Roman' 'Roman (Flavian)' 'Roman (Julio-Cl.)'
 'Roman (Julio-Cl.?)' 'Roman (Severan?)' 'Roman (Augustan)' 'Archaic?'
 'Classical?' 'Hellenistic?' 'Roman- Byzantine' 'Roman (late Antonine'
 'Roman (Neronian?)' 'Roman (Aurelian?)' 'early Hellenistic'
 'Roman (Augustan?)' 'Hellenistic ?' 'Roman (late Severan)'
 'Hellenistic  - Roman' 'Roman (Hadrianic?)' 'late Hellenistic?'
 'Hellenistic  (3rd BC)? Roman (1st - 2nd AD)?' 'Hellenistic? Roman?'
 'Roman (Trajanic?)' 'Roman (Caligula)' 'Roman (Justinian)'
 'late Hellenistic/Roman' 'Classical - Hellenistic 5th - 4th? 3rd? BC'
 'Hellenistic/ Roman' 'Roman? (Severan?)' 'Roman (Trajanic)'
 'Roman (Domitian)' 'Roman (Antonine?)' 'Mycenaean? sub-Mycenaean?'
 'late Archaic - Classical' 'LC

In [ ]:
import numpy as np

periodes_cles = ['Classical', 'Hellenistic', 'Roman', 'Geometric', 'LC', 'Archaic']

df_inscriptions['Date'] = df_inscriptions['Date'].astype(str).replace('nan', '')

def extraire_periode(row_date):
    for p in periodes_cles:
        if p in row_date:
            return pd.Series([p, row_date])
    return pd.Series(["", row_date])
df_inscriptions[['Date', 'Date_Precision_2']] = df_inscriptions['Date'].apply(extraire_periode)
df_inscriptions.loc[df_inscriptions['Date'] == df_inscriptions['Date_Precision_2'], 'Date_Precision_2'] = ""

print(df_inscriptions['Date'].unique())
print(df_inscriptions.value_counts('Date'))

['' 'Classical' 'Archaic' 'LC' 'Hellenistic' 'Roman' 'Geometric']
Date
Roman          1033
Hellenistic     873
                851
Archaic         414
Classical       292
LC              212
Geometric         2
Name: count, dtype: int64


In [ ]:
df_inscriptions.value_counts('Date')

Date
Roman          1033
Hellenistic     873
                851
Archaic         414
Classical       292
LC              212
Geometric         2
Name: count, dtype: int64

In [ ]:
sans_date = df_inscriptions[df_inscriptions['Date'].isin(['', 'nan', None]) | df_inscriptions['Date'].isna()]
sans_date

Inscription #         Type  \
0                                       ICS #456: p.389          CS?   
112                                    ICS #370a: p.422           CS   
115                                     ICS #420: p.373           CS   
119                                    ICS #369c: p.422          EC?   
122   Semitica 32 (1982) 45-49, pl.VII.1 \r= ICS #36...      CS & GA   
...                                                 ...          ...   
3741                                 ICS #248: pp.263-4      CS & GA   
3781                                    Catalogue, p.90  GA \rCS\rPh   
3785                                   ICA 7 (1967) #17          CS?   
3790          BM  97.4-1.711\r=BCH 81 (1957) #2: pp.8-9           CM   
3797                        BCH 81 (1957) #14: pp.19-20           CM   

                                       Place Date  \
0                                  Cyrenaica        
112                         Egypt: Naukratis        
115    Egypt: Thebes \r(Valley of the Kings)        
119                           Greece: Rhodes        
122                         Lebanon: Sarepta        
...                                      ...  ...   
3741  Nicosia: Chytroi\r"Ayios Dhimitrianos"        
3781                                  Cyprus        
3785                                  Cyprus        
3790                       Famagusta: Enkomi        
3797            Nicosia: Apliki "Karamallos"        

                                                 Nature  \
0     No interpretation: ancient (archaizing) imitat...   
112                                  No interpretation.   
115                                  No interpretation.   
119                                  No interpretation.   
122                 Dedication by Timon to \rAsklepios.   
...                                                 ...   
3741                                        Dedication.   
3781                                          Graffiti.   
3785          1 sign.\rPossibly, but probably not, CS.    
3790                                           4 signs.   
3797                                           1 sign.    

                                        Material Place_Region  \
0                             Serpentine scarab.    Cyrenaica   
112                             Terracotta vase.        Egypt   
115                             Rock.\rGraffito.        Egypt   
119                                      Marble.       Greece   
122                                       Stone.      Lebanon   
...                                          ...          ...   
3741                      Marble statuette base.      Nicosia   
3781  Terracotta vases: Greek black-glazed ware.       Cyprus   
3785                          Lead sling bullet.       Cyprus   
3790               Cylinder seal:\rlapis lazuli.    Famagusta   
3797                     Stamp seal: serpentine.      Nicosia   

          Place_precision       Site     Precision_Date  \
0                    None       None                      
112                  None  Naukratis                      
115   Valley of the Kings     Thebes                      
119                  None     Rhodes                      
122                  None    Sarepta                      
...                   ...        ...                ...   
3741   Ayios Dhimitrianos    Chytroi                      
3781                 None       None                      
3785                 None       None                      
3790                 None     Enkomi  signs = secondary   
3797           Karamallos     Apliki                      

             Date_Precision_2  
0                              
112                            
115                            
119                            
122                            
...                       ...  
3741                           
3781                           
3785                           
3790  seal = Mitannia

------------
------------

In [ ]:
df_inscriptions

Inscription # Type  \
0                                       ICS #456: p.389  CS?   
1                                       ICS #419: p.373   CS   
2                                    ICS #392: pp.363-4   CS   
3                                       ICS #388: p.362  EC?   
4                                       ICS #391: p.363   CS   
...                                                 ...  ...   
3800  Enkomi (Schaeffer) 1959 #4\r=Les cylindres-sce...  CM?   
3801        Louvre AM 1639\r=BCH 81 (1957) #9: pp.14-15   CM   
3802       Kourion B1625\r=Bamboula at Kourion, p.144\r  CM?   
3803                K-AD #448\r=K-AD II, p.40, fig.63:6  CM?   
3804                K-AD #449\r=K-AD II, p.40, fig.63:7  CM?   

                                       Place       Date  \
0                                  Cyrenaica              
1      Egypt: Abydos \r(temple of Ramses II)  Classical   
2         Egypt: Abydos \r(temple of Seti I)  Classical   
3         Egypt: Abydos \r(temple of Seti I)  Classical   
4         Egypt: Abydos \r(temple of Seti I)  Classical   
...                                      ...        ...   
3800                       Famagusta: Enkomi         LC   
3801                        Famagusta: Sinda         LC   
3802          Limassol: Kourion \r"Bamboula"         LC   
3803   Larnaca: Kalavasos "Ayios Dhimitrios"         LC   
3804  Larnaca: Kalavasos\r"Ayios Dhimitrios"         LC   

                                                 Nature  \
0     No interpretation: ancient (archaizing) imitat...   
1                          PN: Timos \rson of Thursiya.   
2                        PN: Menokretes \rfrom Salamis.   
3                                  PN: an Eteo-Cypriot?   
4                                                   PN?   
...                                                 ...   
3800                                            1 sign?   
3801                                            1 sign.   
3802                        1 sign? (secondary carving)   
3803                                            1 sign.   
3804                                           1 sign.    

                                           Material Place_Region  \
0                                Serpentine scarab.    Cyrenaica   
1                          Stone wall.  \rGraffito.        Egypt   
2                            Stone wall.\rGraffito.        Egypt   
3                          Stone wall.  \rGraffito.        Egypt   
4                           Stone wall. \rGraffito.        Egypt   
...                                             ...          ...   
3800  Cylinder seal:\rbrown steatite or serpentine.    Famagusta   
3801                       Cylinder seal: hematite.    Famagusta   
3802                      Cylinder seal:\rsteatite.     Limassol   
3803                                  Metal weight.      Larnaca   
3804                                  Metal weight.      Larnaca   

          Place_precision       Site Precision_Date Date_Precision_2  
0                    None       None                                  
1     temple of Ramses II     Abydos      early 4th                   
2        temple of Seti I     Abydos      early 4th                   
3        temple of Seti I     Abydos      early 4th                   
4        temple of Seti I     Abydos      early 4th                   
...                   ...        ...            ...              ...  
3800                 None     Enkomi                           LC 3A  
3801                 None      Sinda                     early LC 3A  
3802             Bamboula    Kourion                           LC 3A  
3803     Ayios Dhimitrios  Kalavasos                           LC 2C  
3804     Ayios Dhimitrios  Kalavasos                           LC 2C  

[3677 rows x 11 columns]

In [ ]:
df_inscriptions.to_csv('cyprus-inscriptions-nettoyees.csv', index=False)

Il faudra donc utiliser Site et Date dans les queries.
Maintenant, on peut passer au géocodage.